esp/public/media/scripts/onsite/teacher_checkin.js

```diff
@@ -112,6 +112,30 @@ $j(function(){
         $msg.text('Checking in...');
     });
 
+    function textTeacher(user, sec, callback, errorCallback){
+        refresh_csrf_cookie();
```

Comment:
Do you really need this?  I don't think anything outside of the CSRF functions itself should need it.